### ISSUE TITLE
Add "select" method to Coordinates object. 

### ISSUE THREAD

**Issue by mpu-creare** [creare-com/podpac#118]
**Description**
I want to be able to select a subset of coordinates by:
* bounding box
* indices

Right now I have to create another set of coordinates and then do an intersection. 

**Describe the solution you'd like**
Basically duplicate the functionality in Coordintes1d for the bounding box part. 
Not sure about the indices... 

**Sample Code**
This is not a Spec, but some notional examples 
```python
coords.select([[min, max], [min, max]])  # user needs to know order of dimensions, this specifies bounds
coords.select(time=[min, max], lat=[min, max])  # Bounds specified
coords.select(time=slice(min_i, max_i, step), lat=slice(min_i, max_i, step))  # Indices
coords[min_i:max_i, min_i:max_i]  # Indices
```

**Comment by jmilloy** [comment]
A similar `select` method should be added to `StackedCoordinates`.

**Comment by jmilloy** [comment]
This is implemented in 6fa9036 along with the RotatedCoordinates feature. Tests still outstanding.
